--- a/build/core/Makefile	2013-05-02 16:11:39.366646305 +0200
+++ b/build/core/Makefile	2013-05-02 16:12:01.795496897 +0200
@@ -1331,8 +1331,6 @@
 	$(OTA_FROM_TARGET_SCRIPT) -v \
 	   -p $(HOST_OUT) \
 	   -k $(KEY_CERT_PAIR) \
-	   --no_assert=$(skip_assert) \
-	   --persist=$(persist) \
 	   --override_device=$(override_device) \
 	   --backup=$(backuptool) \
 	   $(BUILT_TARGET_FILES_PACKAGE) $@
